srml/contract/src/lib.rs

```diff
@@ -14,41 +14,63 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate. If not, see <http://www.gnu.org/licenses/>.
 
-//! Smart-contract module for runtime; Allows deployment and execution of smart-contracts
-//! expressed in WebAssembly.
+//! # Contract Module
+//! 
+//! The contract module provides functionality to deploy and execute smart-contracts expressed in WebAssembly.
+//! To use it in your module, you need to implement the contract [`Trait`].
+//! The supported dispatchable functions are documented as part of the [`Call`] enum.
+//! 
+//! ## Overview
+//! 
+//! This module provides the ability to create smart-contract accounts and send them messages.
+//! A smart-contract is represented by an account with associated code and storage. 
+//! When an account receives a message, the associated code gets executed.
+//! This code can alter the storage entries of the associated account, create new smart-contracts, or send messages to existing smart-contracts.
 //!
-//! This module provides an ability to create smart-contract accounts and send them messages.
-//! A smart-contract is an account with associated code and storage. When such an account receives a message,
-//! the code associated with that account gets executed.
+//! Senders must specify a gas limit with every transaction, as all actions invoked by the smart-contract require gas spend.
+//! Unused gas is refunded after the transaction, regardless of the execution outcome. 
+//! If the gas limit is reached, then all changes made at the specific transaction level are reverted (including balance transfers).
```

Comment:
Good pt - attempt 2: 

**NOTE:** If the gas limit is reached, then all calls and state changes (including balance transfers) are only reverted at the current call's contract level. For example, if contract A calls B and B runs out of gas mid-call, then all of B's dispatched calls are reverted. Assuming correct error handling by contract A, A's other calls and state changes still persist.

Reply:
I'd avoid the term "dispatched calls" here because so far that term means "calls dispatched to the runtime with `ext_dispatch`" which is used for calling the other runtime modules rather than other contracts. (Although all "dispatched calls" are also reverted on failure)